applications/IgaApplication/custom_conditions/sbm_dirichlet_laplacian_condition.cpp

```diff
@@ -0,0 +1,378 @@
+    const auto& r_geometry = this->GetGeometry();
+    const SizeType number_of_nodes = r_geometry.PointsNumber();
+
+    if (rLeftHandSideMatrix.size1() != number_of_nodes || rLeftHandSideMatrix.size2() != number_of_nodes) {
+        rLeftHandSideMatrix.resize(number_of_nodes, number_of_nodes, false);
+    }
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(number_of_nodes, number_of_nodes);
+
+    // Integration
+    const GeometryType::IntegrationPointsArrayType& r_integration_points = r_geometry.IntegrationPoints();
+    const GeometryType::ShapeFunctionsGradientsType& r_DN_De = r_geometry.ShapeFunctionsLocalGradients(r_geometry.GetDefaultIntegrationMethod());
+    
+    // Initialize DN_DX
+    Matrix DN_DX(number_of_nodes,mDim);
+
+    // Differential area
+    double penalty_integration = mPenalty * r_integration_points[0].Weight() ; // * std::abs(DetJ0);
+
+    // Calculating the PHYSICAL SPACE derivatives (it is avoided storing them to minimize storage)
+    noalias(DN_DX) = r_DN_De[0]; // prod(r_DN_De[0],InvJ0);
+
+    Vector H_vec = ZeroVector(number_of_nodes);
+    Matrix DN_dot_n = ZeroMatrix(1, number_of_nodes);
+    Vector DN_dot_n_vec = ZeroVector(number_of_nodes);
+    
+    const Matrix& H = r_geometry.ShapeFunctionsValues();
+    for (IndexType i = 0; i < number_of_nodes; ++i)
+    {
+        H_vec(i) = H(0, i);
+        for (IndexType idim = 0; idim < mDim; idim++) {
+                DN_dot_n_vec(i)  += DN_DX(i, idim) * mNormalPhysicalSpace[idim];         
+        } 
+    }
+    noalias(row(DN_dot_n, 0)) = DN_dot_n_vec;
+
+    // compute Taylor expansion contribution: H_sum_vec
+    Vector H_sum_vec = ZeroVector(number_of_nodes);
+    ComputeTaylorExpansionContribution (H_sum_vec);
+    
+    Matrix H_sum = ZeroMatrix(1, number_of_nodes);
+    noalias(row(H_sum, 0)) = H_sum_vec;
+
+    // Assembly
+    // -(GRAD_w * n, u + GRAD_u * d + ...)
+    noalias(rLeftHandSideMatrix) -= mNitschePenalty * prod(trans(DN_dot_n), H_sum)  * r_integration_points[0].Weight() ; // * std::abs(DetJ0) ;
+    // -(w,GRAD_u * n) from integration by parts -> Fundamental !! 
+    noalias(rLeftHandSideMatrix) -= prod(trans(H), DN_dot_n)                             * r_integration_points[0].Weight() ; // * std::abs(DetJ0) ;
+    // SBM terms (Taylor Expansion) + alpha * (w + GRAD_w * d + ..., u + GRAD_u * d + ...)
+    noalias(rLeftHandSideMatrix) += prod(trans(H_sum), H_sum) * penalty_integration ;
+}
+
+
+void SbmDirichletLaplacianCondition::CalculateRightHandSide(
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    const SizeType mat_size = GetGeometry().size() * 1;
+
+    if (rRightHandSideVector.size() != mat_size)
+        rRightHandSideVector.resize(mat_size);
+    noalias(rRightHandSideVector) = ZeroVector(mat_size);
+
+    MatrixType left_hand_side_matrix = ZeroMatrix(mat_size, mat_size);
+
+    CalculateLocalSystem(left_hand_side_matrix, rRightHandSideVector,
+        rCurrentProcessInfo);

```

Comment:
Considering that this is probably the final implementation, I'd avoid this trick in here.